### ISSUE TITLE
TypeVar of Generic Protocol with mutable field is inferred as covariant

### ISSUE THREAD

**Issue by tamuhey** [microsoft/pyright#2066]
I think Pyright fails to infer the variance of the `TypeVar` of a mutable field in a generic `Protocol`:

```python
from typing import TypeVar, Protocol

T = TypeVar("T")

class P(Protocol[T]):
    v: T # warning: Type variable "T" used in generic protocol "P" should be covariant (reportInvalidTypeVarUse)
```

According to [pep](https://www.python.org/dev/peps/pep-0544/#generic-protocols), a generic `Protocol` with mutable fields is invariant, so I think `T` must be invariant, and no warning or error should be reported.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.156, which I just published. It will also be included in the next release of pylance.
